WalletWasabi/Userfacing/PasswordHelper.cs

```diff
@@ -71,17 +71,10 @@ public static bool IsTooLong(int length)
 		public static bool IsTrimable(string? password, out string? trimmedPassword)
 		{
 			trimmedPassword = password;
-			if (password is null)
-			{
-				return false;
-			}
-
-			var beforeTrim = password.Length;
-
-			trimmedPassword = password.Trim();
 
-			if (beforeTrim != trimmedPassword.Length)
+			if (password.IsTrimable())
 			{
+				trimmedPassword = password!.Trim(); // IsTrimable() returns false if string is null
```

Comment:
This is a hack. Fix Nullable warnings by reconsider what can be null and where. For example, IsTrimable can accept null? That is strange. You removed the null check and put ! there.

Reply:
I did that because am sure at that point the `password` can't be null. `IsTrimable()` extension method accepts null and returns false if the string is null or empty so the `Trim()` never gets called on a null string. So basically I check for null but the compiler doesn't detect that. IMO in this case better to put there the `!` mark instead of a double null check the avoid the warning. But you know better these things, so if you say I put back the null check.

Reply:
you lurked the null check here 

https://github.com/zkSNACKs/WalletWasabi/pull/4676/files#diff-08a0e1ec46bf848bc7d76ebe7ce267b30c8f4403b91bc6437bf41d64de9fd9caR68

Change IsTrimable(this string me) and take out the null check outside to IsTrimable(string? password, out string? trimmedPassword). Then after that password cannot be null.

Reply:
Done